providers/k8s/connection/shared/resources/discovery.go

```diff
@@ -73,7 +73,12 @@ func (d *Discovery) SupportedResourceTypes() (*ApiResourceIndex, error) {
 	log.Debug().Msg("query api resource types")
 	resList, err := d.discoveryClient.ServerPreferredResources()
 	if err != nil {
-		return nil, errors.Wrap(err, "failed to fetch api resource types from kubernetes")
+		var groupDiscoveryErr *discovery.ErrGroupDiscoveryFailed
+		if errors.As(err, groupDiscoveryErr) {
+			log.Warn().Err(err).Msg("one or more kubernetes API groups fail to load")
```

Comment:
nit: I think this one is not necessarily a warning. I would log it as a debug message. 

Also you can do:
```go
if discovery.IsGroupDiscoveryFailedError(err) {
  ...
}
```